pkg/providers/base.go

```diff
@@ -0,0 +1,50 @@
+package providers
+
+import (
+	"log"
+)
+
+type Config struct {
+	BaseProvider
+	GoogleApps
+}
+
+type IProvider interface {
+	Finalise() error
+	IsSetCorrectly() (bool, error)
+
+	GetClientID() string
+	GetClientSecret() string
+	GetCallbackURL() string
```

Comment:
Dont think these 3 belong in the provider interface, if we're going with the design of 

~/dexy --config=/dexy/google.yaml 

then callback_url, client_id and client_secret will all be top level config values